test/k8sT/Policies.go

```diff
@@ -808,3 +815,284 @@ var _ = Describe("K8sValidatedPolicyTestAcrossNamespaces", func() {
+	Expect(err).NotTo(HaveOccurred(), "Pod did not finish as expected.")
+	Expect(success).To(BeTrue(), "Unable to connect to service %s on port %d. (It should)", service.String(), dPort)
+}
+
+// testCannotConnect creates and tests if a given pod can not connect to a given
+// service on a specific destination port.
+func testCannotConnect(k *helpers.Kubectl, ns, podName string, service *v1.Service, dPort int) {
+	pod := createNetworkClientPod(k, ns, podName, service, dPort)
+	defer func() {
+		By(fmt.Sprintf("Cleaning up the pod %s", podName))
+		err := k.CoreV1().Pods(ns).Delete(pod.Name, nil)
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have been deleted", pod.Name))
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	err := k.WaitForPodReady(ctx, ns, podName)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have been ready", pod.Name))
+
+	ctx, cancel = context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	err = k.WaitForPodFinish(ctx, ns, podName)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have finish successfully", pod.Name))
+
+	ctx, cancel = context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	success, err := k.WaitForPodSuccess(ctx, ns, podName)
+	Expect(err).NotTo(HaveOccurred(), "Pod did not finish as expected.")
+	Expect(success).To(BeFalse(), "Able to connect to service %s on port %d. (It shouldn't)", service.String(), dPort)
+}
+
+func cleanupNetworkPolicy(k *helpers.Kubectl, policy *networkingv1.NetworkPolicy) {
+	By("Cleaning up the policy.")
+	err := k.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, nil)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+var _ = Describe("K8sValidatedPolicyTestNamespaceSelector", func() {
+	var service *v1.Service
+	var podServer *v1.Pod
+	var kub *helpers.Kubectl
+
+	logger := log.WithFields(logrus.Fields{"testName": "K8sValidatedPolicyTestNamespaceSelector"})
+	logger.Info("Starting")
+
+	ns := "namespace-selector-test"
+
+	Context("KubernetesNetworkPolicy between server and client", func() {
+		BeforeEach(func() {
+			kub = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+
+			By("Creating the namespace that will be used for the pods")
+			_, err := kub.CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns}})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating a simple server that serves on port 80 and 81.")
+			podServer, service = createServerPodAndService(kub, ns, "server", []int{80, 81})
+
+			By("Waiting for pod ready", func() {
+				ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+				defer cancel()
+				err := kub.WaitForPodReady(ctx, ns, podServer.Name)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			By("Testing pods can connect to both ports when no policy is present.")
+			testCanConnect(kub, ns, "client-can-connect-80", service, 80)
```

Comment:
As @ianvernon mention, use testConnect and send BeFalse() or BeTrue() as argument.